contributors/design-proposals/auditing.md

```diff
@@ -0,0 +1,340 @@
+    Level Level
+
+    // The users (by authenticated user name) this rule applies to.
+    // An empty list implies every user.
+    Users []string
+    // The user groups this rule applies to. If a user is considered matching
+    // if the are a member of any of these groups
+    // An empty list implies every user group.
+    UserGroups []string
+
+    // The verbs that match this rule.
+    // An empty list implies every verb.
+    Verbs []string
+
+    // Rules can apply to API resources (such as "pods" or "secrets"),
+    // non-resource URL paths (such as "/api"), or neither, but not both.
+    // If neither is specified, the rule is treated as a default for all URLs.
+
+    // APIGroups is the name of the APIGroup that contains the resources ("" for core).
+    // If multiple API groups are specified, any action requested against one of the
+    // enumerated resources in any API group will be allowed.
+    // Any empty list implies every group.
+    APIGroups []string
+    // Namespaces that this rule matches.
+    // This field should be left empty if specifying non-namespaced resources.
+    // Any empty list implies every namespace.
+    Namespaces []string
+    // Resources is a list of resource types this rule applies to.
+    // Any empty list implies every resource type.
+    Resources []string
+    // ResourceNames is an optional white list of names that the rule applies to.
+    // Any empty list implies everything.
+    ResourceNames []string
+
+    // NonResourceURLs is a set of partial urls that should be audited.
+    // *s are allowed, but only as the full, final step in the path.
+    // If an action is not a resource API request, then the URL is split on '/' and
+    // is checked against the NonResourceURLs to look for a match.
+    NonResourceURLs []string
+}
+```
+
+As an example, the administrator may decide that by default requests should be audited with the
+response, except for get and list requests. On top of that, they wish to completely ignore the noisy
+`kube-proxy` endpoint requests. This looks like:
+
+```yaml
+rules:
+    - level: None
+      users: ["system:kube-proxy"]
+      apiGroups: [""] # The core API group
+      resources: ["endpoints"]
+    - level: RequestBody
+      verbs: ["get", "list"]
+    # The default for non-resource URLs
+    - level: Metadata
+      nonResourceURLs: ["*"]
+    # The default for everything else
+    - level: ResponseBody
+```
+
+The policy is checked immediately after authentication in the request handling, and determines how
+the `audit.Event` is formed.
+
+### Filters
+
+In addition to the high-level policy rules, auditing can be controlled at a more fine-grained level
+with `Filters`. Unlike the policy, filters are applied _after_ the `audit.Event` is constructed, but
+before it's passed to the output backend.
+
+TODO: Define how filters work. They should enable dropping sensitive fields from the
```

Comment:
See my other [comment](https://github.com/kubernetes/community/pull/145/files#r113126633). Basically I've talked with @timstclair and we agreed it'll be a separate proposal, we need to update this here accordingly.